csatar-plugins/moodleSideCode/csatar.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+// This file should be copied to the moodle root directory.
+// Moodle config should have a $CFG->csatarEncryptionKey property with the correct encryption key set.
+
+require('config.php');
+
+$receivedData = prepareReceivedData();
+
+$userDataFromDb = getUserData($receivedData);
+
+$userId = null;
+
+if ($userDataFromDb === false) {
+    $userId = createNewUser($receivedData['basic']);
+} else {
+    $userId = intval($userDataFromDb->id);
+}
+
+$user = get_complete_user_data('id', $userId);
+
+// require_once($CFG->dirroot."/user/lib.php");
+// user_delete_user($user); die;
+
+if (!empty($user)) {
+    // update user data
+    updateBasicData($user, $receivedData['basic']);
+
+    // update additional data
+    updateAdditionalData($user->id, $receivedData['profile']);
+}
+
+loginAndRedirect($user);
+
+// functions
+
+function createNewUser($receivedData) {
+    // check if user with same e-mail address exists
+    if (\core_user::get_user_by_email($receivedData['email']) !== false) {
+        echo "User already exists with the same e-mail address, please contact system administrator!";
+        die;
+    }
+
+    global $CFG;
+    require_once($CFG->dirroot."/user/lib.php");
+
+    // create user object
+    $newUser = (object) $receivedData;
+    $newUser->username  = generateUsername($receivedData);
+    $newUser->auth      = 'manual';
+    $newUser->confirmed = 1;
+
+    return $userId = user_create_user($newUser, false, false);
+}
+
+function generateUsername($receivedData, $num = 1) {
+    if ($num > 10) {
```

Comment:
What if there are more than 10 users with the same name?
```suggestion
    if ($num > 999) {
```